src/Medidata.ZipkinTracer.Core/ZipkinClient.cs

```diff
@@ -122,6 +120,55 @@ public void EndServerTrace(Span serverSpan)
                 {
                     logger.Error("Error Ending Server Trace", ex);
                 }
+        }
+
+        /// <summary>
+        /// Records an annotation with the current timestamp and the provided value in the span.
+        /// </summary>
+        /// <param name="span">The span where the annotation will be recorded.</param>
+        /// <param name="value">The value of the annotation to be recorded. If this parameter is omitted
+        /// (or its value set to null), the method caller member name will be automatically passed.</param>
+        public void Record(Span span, [CallerMemberName] string value = null)
+        {
+            if (!isTraceOn)
+                return;
+
+            try
+            {
+                spanTracer.Record(span, value);
+            }
+            catch (Exception ex)
+            {
+                logger.Error("Error recording the annotation", ex);
+            }
+        }
+
+        /// <summary>
+        /// Records a key-value pair as a binary annotiation in the span.
+        /// </summary>
+        /// <typeparam name="T">The type of the value to be recorded. See remarks for the currently supported types.</typeparam>
+        /// <param name="span">The span where the annotation will be recorded.</param>
+        /// <param name="key">The key which is a reference to the recorded value.</param>
+        /// <param name="value">The value of the annotation to be recorded.</param>
+        /// <remarks>The RecordBinary will record a key-value pair which can be used to tag some additional information
+        /// in the trace without any timestamps. The currently supported value types are <see cref="bool"/>,
+        /// <see cref="byte[]"/>, <see cref="short"/>, <see cref="int"/>, <see cref="long"/>, <see cref="double"/> and
+        /// <see cref="string"/>. Any other types will be passed as string annotation types.
+        /// 
+        /// Please note, that although the values have types, they will be recorded and sent by calling their
+        /// respective ToString() method.</remarks>
+        public void RecordBinary<T>(Span span, string key, T value)
+        {
+            if (!isTraceOn)
+                return;
+
+            try
+            {
+                spanTracer.RecordBinary<T>(span, key, value);
+            }
+            catch (Exception ex)
+            {
+                logger.Error($"Error recording a binary annotation (key: {key})", ex);
```

Comment:
I built and ran Ampridatvir with VS2013 and the ZipkinTracerModule built in VS2015 and there were no errors, so this change should impact the development of the ZipkinTracerModule only. At runtime or the development of other applications which referencing this should not have any problems even if VS2013 is used for them.